plugins/scaffolder-backend/src/scaffolder/stages/prepare/github.ts

```diff
@@ -36,13 +42,15 @@ export class GithubPreparer implements PreparerBase {
       parsedGitUrl.filepath ?? '',
     );
 
-    const git = this.config.token
-      ? Git.fromAuth({
-          username: 'x-access-token',
-          password: this.config.token,
-          logger,
-        })
-      : Git.fromAuth({ logger });
+    const { token } = await this.config.credentialsProvider.getCredentials({
+      url,
+    });
+
+    const git = Git.fromAuth({
+      username: 'x-access-token',
```

Comment:
Prolly best to keep the logic of creating the instance with `Git.fromAuth({ logger })` if we don't get a token